### PR TITLE
Adjust zones save/load cycle

### PR DESCRIPTION
**Pull request by Menschheit**
#### Summary
Bugfixes "Improve zones save/load cycle"

#### Purpose of change

1. If `<playername>.zones.json ` doesn't exist, `zone_manager::zones` will not be updated by `load_zone`, that is the cause of #51099.
2. All zones are saved in the same file:  `<playername>.zones.json`, which can only be updated if the zones manager UI is called, resulting #49755.
3. The zones are saved only if the player has made some changes with zones manager, it may lose synchronization with the game state(If the player quits without saving).

#### Describe the solution
1. If there is no file to load, clear zones.
2. Save world zones( non-"your follower"-factions) in a different file independent of the player name as soon as they are added.
3. Store all zones changes in temporary files. The zones are only saved while saving the game. 

#### Describe alternatives you've considered

Doing termporary changes in RAM,

#### Testing

limited testing of the situations mentioned by those two issues, worked as intended.

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
> #### Describe alternatives you've considered
> 
> Doing termporary changes in RAM,

Why not? What is gained by using temporary files? Surely only reading and writing the zones when the game is saved/loaded would simplify the code.

**Comment by Menschheit** [comment]
> > #### Describe alternatives you've considered
> > Doing termporary changes in RAM,
> 
> Why not? What is gained by using temporary files? Surely only reading and writing the zones when the game is saved/loaded would simplify the code.

Because... I am lazy? In this way I don't need to touch `game::zones_manager()` at all.

To do it in RAM, first I need to make a copy of `zone_manager::zones` which is a private member, it means I need to write another function to do the job, then I need to track every call of member functions which has access to `zone_manager::zones`  and redirect them to my copy.

Since it is originally done with saving/loading files, doing it in RAM requires some extra work.

I am not a programmer just a player who wants to solve problems while devs are too busy. I don't know anything about performance/ RAM management/ programming style etc.

So basically I tend to do minimal changes to get things work( I really appreciate the devs who do alot of migrations and restructures to improve the codes) and hope other ppl are ok with it too.

**Comment by I-am-Erk** [comment]
@kevingranade Is there a reason not to merge this? It's passing all tests and fixes a bug.

File: src/clzones.cpp [hunk 1]
```diff
@@ -1049,9 +1052,16 @@ void zone_manager::add( const std::string &name, const zone_type_id &type, const
 
     //Create a regular zone
     zones.push_back( new_zone );
+
+  //If not player defined, save them immediately
+    if( fac != faction_id( "your_followers" ) ) {
+        save_world_zones();
+    }
+
```
Comment (ZhilkinSerg) [inline src/clzones.cpp:1060]:
```suggestion

```

File: src/clzones.cpp [hunk 2]
```diff
@@ -1049,9 +1054,16 @@ void zone_manager::add( const std::string &name, const zone_type_id &type, const
 
     //Create a regular zone
     zones.push_back( new_zone );
+
+  //If not player defined, save them immediately
+    if( fac != faction_your_followers ) {
+        save_world_zones();
+    }
+
```
Comment (ZhilkinSerg) [inline src/clzones.cpp:1062]:
```suggestion

```

File: src/clzones.cpp [hunk 3]
```diff
@@ -1049,9 +1054,15 @@ void zone_manager::add( const std::string &name, const zone_type_id &type, const
 
     //Create a regular zone
     zones.push_back( new_zone );
+
+  //If not player defined, save them immediately
```
Comment (ZhilkinSerg) [inline src/clzones.cpp:1058]:
```suggestion
    //If not player defined, save them immediately
```
